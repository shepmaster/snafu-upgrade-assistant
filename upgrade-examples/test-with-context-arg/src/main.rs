use snafu::{Snafu, ResultExt};

#[derive(Debug, Snafu)]
struct Inner;

#[derive(Debug, Snafu)]
enum Error {
    Variant1 { source: Inner },
}

fn main() {
    let inner = InnerContext.fail::<()>();
    let _ = inner.with_context(|| Variant1);
}
