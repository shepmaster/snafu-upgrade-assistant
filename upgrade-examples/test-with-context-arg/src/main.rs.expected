use snafu::{Snafu, ResultExt};

#[derive(Debug, Snafu)]
struct Inner;

#[derive(Debug, Snafu)]
enum Error {
    Variant1 { source: Inner },
}

fn main() {
    let inner = InnerSnafu.fail::<()>();
    let _ = inner.with_context(|_| Variant1Snafu);
}
